lib/system/dyncalls.nim

```diff
@@ -149,6 +149,19 @@ elif defined(windows) or defined(dos):
       if result != nil: return
     procAddrError(name)
 
+  proc MessageBoxA(hWnd: cint, lpText, lpCaption: cstring, uType: int): int32 {.
+    header: "<windows.h>", nodecl.}
+
+  proc nimLoadLibraryError(path: string) =
+    when defined(guiapp):
+      discard MessageBoxA(0, msg, "could not load: " & path, 0)
```

Comment:
This uses ``&``  which uses the GC. Boehm GC is loaded as a DLL --> crash if Boehm cannot be loaded. You need to use some low level stuff to fill an array[Size, char] and pass that to MessageBoxA. Also MessageBoxA is already declared in ``excpt.nim``.